packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts

```diff
@@ -550,7 +546,17 @@ export class PostgresClient<
 }
 
 // Mimics behaviour in https://github.com/brianc/node-postgres/blob/master/packages/pg-connection-string/index.js
-const getSSLConfigFromMode = (mode: string): PoolConfig['ssl'] => {
+const getSSLConfigFromMode = ({
+    sslcert,
+    sslkey,
+    sslmode,
+    sslrootcert,
+}: SslConfiguration): PoolConfig['ssl'] => {
+    const mode = sslmode || 'prefer';
+    const ca = sslrootcert || [
+        ...rootCertificates,
+        readFileSync(path.resolve(__dirname, './ca-bundle-aws-rds-global.pem')),
```

Comment:
This sync call could become a problem, maybe instead of `PostgresWarehouseClient` accepting a path to cert as an argument it should accept file contents directly, so that the function initializing the class would be responsible for reading a file and that could be async?